motor/rule_tests/owasp_tests/no_additional_properties_test.go

```diff
@@ -0,0 +1,100 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/daveshanley/vacuum/model"
+	"github.com/daveshanley/vacuum/motor"
+	"github.com/daveshanley/vacuum/rulesets"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRuleSet_OWASPNoAdditionalProperties_Success(t *testing.T) {
+
+	tc := []struct {
+		name string
+		yml  string
+	}{
+		{
+			name: "valid case: oas2 does not allow additionalProperties by default so dont worry about it",
+			yml: `swagger: "2.0"
+info:
+  version: "1.0"
+definitions:
+  Foo:
+    type: object
+    additionalProperties: false
+`,
+		},
+		{
+			name: "valid case: oas3",
+			yml: `openapi: "3.0.0"
+info:
+  version: "1.0"
+components:
+  schemas:
+    Foo:
+      type: object
+      additionalProperties: false
+`,
+		},
+		{
+			name: "valid case: no additionalProperties defined",
+			yml: `openapi: "3.0.0"
+info:
+  version: "1.0"
+components:
+  schemas:
+    Foo:
+      type: object
+`,
+		},
+	}
+
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			rules := make(map[string]*model.Rule)
+			rules["owasp-no-additionalProperties"] = rulesets.GetOWASPNoAdditionalPropertiesRule()
+
+			rs := &rulesets.RuleSet{
+				Rules: rules,
+			}
+
+			rse := &motor.RuleSetExecution{
+				RuleSet: rs,
+				Spec:    []byte(tt.yml),
+			}
+			results := motor.ApplyRulesToRuleSet(rse)
+			assert.Len(t, results.Results, 0)
+		})
+	}
+}
+
+func TestRuleSet_OWASPNoAdditionalProperties_Error(t *testing.T) {
```

Comment:
Shouldn't falsy also see if it is of type object and if it not nil? I am not sure how to do this without having another Then :/

Reply:
![image](https://github.com/daveshanley/vacuum/assets/33238205/5688b053-4c32-4623-a500-01de34f3d9e8)
I believe falsy should also check if the content is empty, right? This is not the case right now :/

Reply:
it should yes, I guess that was an oversight of mine.

Reply:
Changed the falsy function to warn whenever a node does not have an empty content (meaning that if the additionalProperties would evaluate to an object != nil then it would be true and an error would ocur). Will create a small test shortly